apis/feature_extraction/training/train_lsh.py

```diff
@@ -0,0 +1,56 @@
+import os
+import pickle
+
+from datetime import datetime
+import click
+import numpy as np
+
+from src.aws import get_keys_and_ids, get_object_from_s3, put_object_to_s3
+from src.lsh import split_features, train_clusters
+from src.elastic import get_random_feature_vectors
+
+
+def load_local_feature_vectors(feature_vector_path, sample_size):
+    ids = np.random.choice(
+        os.listdir(feature_vector_path),
+        size=sample_size,
+        replace=False
+    )
+
+    feature_vectors = []
+    for id in ids:
+        with open(os.path.join(feature_vector_path, id)) as f:
+            feature_vectors.append(np.fromfile(f, dtype=np.float32))
+
+    return np.stack(feature_vectors)
+
+
+@click.command()
+@click.option('-n', help='number of groups to split the feature vectors into', default=256)
+@click.option('-m', help='number of clusters to find within each feature group', default=32)
+@click.option('--sample_size', help='number of embeddings to train clusters on', default=25_000)
+@click.option('--feature_vector_path', help='path to a synced local version of the fvs in s3')
+def main(n, m, sample_size, feature_vector_path):
+    if feature_vector_path:
```

Comment:
This is here to kick off the model-trainer <--> inferrer loop - the model trainer needs to get a set of feature vectors from somewhere, and it can't look for them in an index which hasn't been populated yet. I imagine this would need to be run locally to get an initial model into s3, at which point the loop can continue in AWS? Shouldn't need to be rerun unless disaster strikes, but it should definitely be documented if it's going to stay in the code - will make that change now.

Reply:
Scratch that - moving the initial training and dumping into s3 into a separate script, disentangled from the production logic